src/types.ts

```diff
@@ -117,3 +117,5 @@ export type SignedTransaction = any;
 
 // TODO
 export type SignedCanceledTransaction = any;
+
+export type Hex = `0x${string}`;
```

Comment:
Could this be imported from `@metamask/utils`?

Reply:
I suppose we could. I assume this is used somewhere else in the extension dependency graph so probably not adding any weight. I just feel silly adding the full package just to pull in a type here...

Reply:
cc @mcmire how do you generally think about these cases ☝️ ?

Reply:
Drive by comment: `0x{string}` isn't Hex, and `@metamask/utils` is used by lots of other packages, so this shouldn't be in a general utility library if its is to be named `Hex`. Unbounded (length) hex strings are currently not possible in TypeScript (https://github.com/microsoft/TypeScript/issues/43335).

You can maybe get typescript to full check 30ish characters or so, more if you make use of a "trampoline", but type checking will be so slow it'll be unusable.

Reply:
That's true — the `Hex` type isn't as useful as it could be. For what it's worth we do have [some ways to double check that a string is a real hex string at runtime in `@metamask/utils`](https://github.com/MetaMask/utils/blob/8732f000b7f880cc6d407b23a1f21522dae1b8fc/src/hex.ts), although we might not be using them as much as we should. I tend to think of `Hex` as more of a signal to readers than an actual constraint for consumers.

Since it's basically an approximation and easy to write, I'm okay with copying this and not importing it from `@metamask/utils`. If we need something else from utils then we can always bring the package in later.